packages/@n8n/db/src/repositories/settings.repository.ts

```diff
@@ -1,10 +1,10 @@
-import { Settings } from '@n8n/db';
 import { Service } from '@n8n/di';
 import { DataSource, Repository } from '@n8n/typeorm';
 import { ErrorReporter } from 'n8n-core';
 
-import config from '@/config';
-import { EXTERNAL_SECRETS_DB_KEY } from '@/external-secrets.ee/constants';
+import { Settings } from '../entities';
+
+const EXTERNAL_SECRETS_DB_KEY = 'feature.externalSecrets';
```

Comment:
I feel like this reverses the inversion of control (settings delegate to external secrets to know in which key to save the data). 
Would it make sense to instead remove the dedicated external secrets settings repository method (like `saveEncryptedSecretsProviderSettings` and `getEncryptedSecretsProviderSettings`) and have the external-secrets-manager saves/get the data using general `save` / `find` method of the settings repository, to be in control of what it should read / write to the settings ?

Reply:
I like this, but this settings-secrets setup exists like this in `master` and this PR is strictly only for moving - I hesitate to make logic changes here unless I really have to, as with `dismissBanner`.

Reply:
Make sense, it's not really related to your PR, the issue was already there before (external-secrets calls settings which itself calls external secrets for constant)